src/python_testing/TC_AVSM_2_8.py

```diff
@@ -0,0 +1,206 @@
+            endpoint=endpoint, cluster=cluster, attribute=attr.RateDistortionTradeOffPoints
+        )
+        logger.info(f"Rx'd RateDistortionTradeOffPoints: {aRateDistortionTradeOffPoints}")
+        aMinViewport = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.MinViewport
+        )
+        logger.info(f"Rx'd MinViewport: {aMinViewport}")
+        aVideoSensorParams = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.VideoSensorParams
+        )
+        logger.info(f"Rx'd VideoSensorParams: {aVideoSensorParams}")
+        aMaxEncodedPixelRate = await self.read_single_attribute_check_success(
+            endpoint=endpoint, cluster=cluster, attribute=attr.MaxEncodedPixelRate
+        )
+        logger.info(f"Rx'd MaxEncodedPixelRate: {aMaxEncodedPixelRate}")
+
+        try:
+            asserts.assert_greater(len(aRankedStreamPriorities), 0, "RankedVideoStreamPrioritiesList is empty")
+            asserts.assert_greater(len(aRateDistortionTradeOffPoints), 0, "RateDistortionTradeOffPoints is empty")
+            videoStreamAllocateCmd = commands.VideoStreamAllocate(
+                streamUsage=aRankedStreamPriorities[0],
+                videoCodec=aRateDistortionTradeOffPoints[0].codec,
+                minFrameRate=aVideoSensorParams.maxFPS,
+                maxFrameRate=aVideoSensorParams.maxFPS,
+                minResolution=aRateDistortionTradeOffPoints[0].resolution,
+                maxResolution=aRateDistortionTradeOffPoints[0].resolution,
+                minBitRate=aRateDistortionTradeOffPoints[0].minBitRate,
+                maxBitRate=aRateDistortionTradeOffPoints[0].minBitRate,
+                minFragmentLen=4000,
+                maxFragmentLen=4000,
+            )
+            await self.send_single_cmd(endpoint=endpoint, cmd=videoStreamAllocateCmd)
+        except InteractionModelError as e:
+            asserts.assert_equal(e.status, Status.Success, "Unexpected error returned")
+            pass
+
+    @async_test_body
+    async def test_TC_AVSM_2_8(self):
+        endpoint = self.get_endpoint(default=1)
+        cluster = Clusters.CameraAvStreamManagement
+        attr = Clusters.CameraAvStreamManagement.Attributes
+        commands = Clusters.CameraAvStreamManagement.Commands
+
+        self.step("precondition")
+        # Commission DUT - already done
+        await self._precondition_one_allocated_vdo_stream()
+
+        self.step(1)
+        aFeatureMap = await self.read_single_attribute_check_success(endpoint=endpoint, cluster=cluster, attribute=attr.FeatureMap)
+        logger.info(f"Rx'd FeatureMap: {aFeatureMap}")
+        vdoSupport = (aFeatureMap & cluster.Bitmaps.Feature.kVideo) > 0
+        wmarkSupport = (aFeatureMap & cluster.Bitmaps.Feature.kWatermark) > 0
+        osdSupport = (aFeatureMap & cluster.Bitmaps.Feature.kOnScreenDisplay) > 0
+        asserts.assert_true(
```

Comment:
This shouldn't fail if neither watermark nor OSD are supported, it just shouldn't be run at all.  
Handle that as a decorator along the lines of:
@run_if_endpoint_matches(has_feature(Clusters.CameraAvStreamManagement,                                         Clusters.CameraAvStreamManagement.Bitmaps.Feature.kWatermark) or
                             has_feature(Clusters.CameraAvStreamManagement, Clusters.CameraAvStreamManagement.Bitmaps.Feature.kOnScreenDisplay))

Reply:
Thank you for your comments. We fixed it with our new commit.

Reply:
Okay, Thank you for your comments. so, we need to add WMARK or OSD in camera app or we need to remove it from this patch? Which is better?

Reply:
Have the testcase only run if WMARK or OSD are present; which is what the @run_if_endpoint I posted above does.